cli/command/image/trust.go

```diff
@@ -296,7 +298,10 @@ func trustedPull(ctx context.Context, cli command.Cli, repoInfo *registry.Reposi
 }
 
 // imagePullPrivileged pulls the image and displays it to the output
-func imagePullPrivileged(ctx context.Context, cli command.Cli, authConfig types.AuthConfig, ref string, requestPrivilege types.RequestPrivilegeFunc, all bool) error {
+func imagePullPrivileged(ctx context.Context, cli command.Cli, authConfig types.AuthConfig, ref string, requestPrivilege types.RequestPrivilegeFunc, all bool, platform specs.Platform) error {
+	if platform.OS == "" && os.Getenv("DOCKER_DEFAULT_PLATFORM") != "" {
+		platform.OS = os.Getenv("DOCKER_DEFAULT_PLATFORM")
```

Comment:
This should be handled somewhere else before it gets into the pull handling. Possibly as part of the `PlatformOpt` type. It can be used as a default value for `Value()`